MHRA/products
# Substance page uses API

# Points the substance page to the API

Substance page using API. Hidden by feature switch for now, see #693.

![](https://media.giphy.com/media/y5OS4cGeLMQsU/giphy.gif)

### Acceptance Criteria

- [ ] Substance page uses API, passing a substance name and receiving a list of products
- [ ] Caching of requests, so users don't fire the same graphql query twice in a session

### Testing information

1. Run `make` in the `medicines/api` directory.
2. Run `yarn dev` in the `medicines/web` directory.
3. Navigate to `localhost:3000` and go to a substance page via the index.
4. See the request in your browser's network dev tools. Notice that it's a graphql request to `localhost:8000`.

### Pre-flight Checklist


### Pre-merge Checklist


[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"
